python/graphstorm/model/utils.py

```diff
@@ -601,6 +697,172 @@ def save_hdf5_embeddings(emb_path, embeddings, rank, world_size,
+        #     - owner of the folder can read, write, and execute;
+        #     - owner' group can read, write;
+        #     - others can read, write, and execute.
+        os.chmod(save_embed_path, 0o767)
+        logging.info("Writing GNN embeddings to "\
+            "%s in pytorch format.", save_embed_path)
+
+    # make sure the save_embed_path permission is changed before other process start to save
+    barrier()
+
+    emb_info = {
+        "format": "pytorch",
+        "emb_name":[],
+        "world_size":world_size
+    }
+
+    for ntype, (embs, nids) in shuffled_embs.items():
+        os.makedirs(os.path.join(save_embed_path, ntype), exist_ok=True)
+        assert len(nids) == len(embs), \
+            f"The embeding length {len(embs)} does not match the node id length {len(nids)}"
+        th.save(embs, os.path.join(os.path.join(save_embed_path, ntype),
+                                  f'emb.part{pad_file_index(rank)}.bin'))
+        th.save(nids, os.path.join(os.path.join(save_embed_path, ntype),
+                                  f'nids.part{pad_file_index(rank)}.bin'))
+        emb_info["emb_name"].append(ntype)
+
+    if rank == 0:
+        with open(os.path.join(save_embed_path, "emb_info.json"), 'w', encoding='utf-8') as f:
+            json.dump(emb_info, f, indent=4)
+
+def save_full_node_embeddings(g, save_embed_path,
+                              embeddings,
+                              node_id_mapping_file,
+                              save_embed_format="pytorch"):
+    """ Save all node embeddings with node IDs in Graph Node ID space.
+
+        The input node `embeddings` are stored in Partition Node ID space.
+        By default, `save_full_node_embeddings` will translate the node IDs
+        from Partition Node ID space into their counterparts in Graph Node
+        ID space.
+
+        `save_full_node_embeddings` will save two information of an
+        embedding: 1) the embedding and 2) its corresponding node ID
+        in Graph Node ID space.
+        It assumes the input `embeddings` are stored in Partition Node
+        ID space and the IDs start from 0 to N. It will call NodeIDShuffler
+        to shuffle the node IDs from Partition Node ID space into Graph
+        Node ID space.
+
+        The saved node embeddings are in the following format:
+
+        Example
+        --------
+        # embedddings:
+        #   ntype0:
+        #     nids.part00000.bin
+        #     nids.part00001.bin
+        #     ...
+        #     emb.part00000.bin
+        #     emb.part00001.bin
+        #     ...
+        #   ntype1:
+        #     nids.part00000.bin
+        #     nids.part00001.bin
+        #     ...
+        #     emb.part00000.bin
+        #     emb.part00001.bin
+        #     ...
+
+        The content of nids.part files and emb.part files looks like:
+
+        Example:
+        --------
+
+        .. code::
+        nids.part00000.bin   |   emb.part00000.bin
```

Comment:
1. add a space line after `.. code::`;
2. add indent to each line of the contents.